src/uuid/rfc4122.cr

```diff
@@ -0,0 +1,85 @@
+# Support for RFC 4122 UUID variant.
+struct UUID
+  # RFC 4122 UUID variant versions.
+  enum Version
```

Comment:
Could we get rid of this enum? Since it's just a number, why not just deal with a Int32? @asterite ?